src/transformState.js

```diff
@@ -4,8 +4,21 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      Object.assign(state, key.extraData);
+    } else if (key.type === 'clear') {
+      for (const prop of Object.getOwnPropertyNames(state)) {
+        delete state[prop];
+      }
+    } else if (key.type === 'removeProperties') {
+      for (const val of key.keysToRemove) {
+        delete state[val];
+      }
+    }
```

Comment:
Try to avoid else if statements.
Code not good readable with them.
Better to use switch...case here